src/store/facade/GroceryStore.java

```diff
@@ -769,4 +779,55 @@ public Result processShipment(Request request) {
 		return result.iterator();
 	}
 
+	/**
+	 * Saves the GroceryStore object to file BACKUP_FILE_NAME in current directory,
+	 * including static fields memberIdCounter and orderIdCounter.
+	 * 
+	 * @param groceryStore - GroceryStore object being saved.
+	 * @return TRUE if file was successfully saved, FALSE otherwise
+	 * @throws Exception for any problems opening or writing the file throws an
+	 *                   Exception and returns FALSE
+	 */
+	public static boolean save(GroceryStore groceryStore) throws Exception {
+		try {
+			FileOutputStream file = new FileOutputStream(BACKUP_FILE_NAME);
+			ObjectOutputStream object = new ObjectOutputStream(file);
+			object.writeObject(groceryStore);
+			object.writeObject(memberIdCounter);
+			object.writeObject(orderIdCounter);
```

Comment:
Haven't worked with it but I'm assuming the memberIdCounter and orderIdCounter are not carried within the groceryStore object?

Reply:
My other comment should answer that. (GroceryStore is exactly where they are.)

Reply:
Maybe I wasn't clear, I mean the lines with: 
			object.writeObject(groceryStore);
			object.writeObject(memberIdCounter);
			object.writeObject(orderIdCounter);
We are already writing the groceryStore object, wouldn't that contain the id counters as fields?